src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/NumberFilter.php

```diff
@@ -0,0 +1,151 @@
+                $clause = [
+                    'range' => [
+                        $attributePath => ['lt' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::LOWER_OR_EQUAL_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['lte' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::EQUALS:
+                $clause = [
+                    'term' => [
+                        $attributePath => $value
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::NOT_EQUAL:
+                $mustNotClause = [
+                    'term' => [
+                        $attributePath => $value,
+                    ],
+                ];
+                $filterClause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+            case Operators::GREATER_OR_EQUAL_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['gte' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::GREATER_THAN:
+                $clause = [
+                    'range' => [
+                        $attributePath => ['gt' => $value]
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath
+                    ]
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath
+                    ]
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Checks that the value is a number.
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $value
+     */
+    protected function checkValue(AttributeInterface $attribute, $value)
+    {
+        if (!is_numeric($value) && null !== $value) {
```

Comment:
I agree with @momoss 
I don't understand why we moved all those check methods in a helper
at least we should have introduced a DataChecker, it would have been more precise...